views: search: Default order latest grants by date unless text search

michaelwood opened the pull request:
Fixes: https://github.com/ThreeSixtyGiving/grantnav/issues/1193

Commit:
views: search: Default order latest grants by date unless text search

Fixes: https://github.com/ThreeSixtyGiving/grantnav/issues/1193

michaelwood commented:
Now available to test on dev.

Tests cases are broken because of reliance of search order, will fix once intended sort order behaviour is confirmed.

mariongalley commented:
@michaelwood Had a go at testing on dev and couldn't get it to load - this is the /search page

<img width="1069" height="936" alt="image" src="https://github.com/user-attachments/assets/709a2bf3-bf9d-4d98-88ce-3705c21b4833" />

michaelwood commented:
> @michaelwood Had a go at testing on dev and couldn't get it to load - this is the /search page
> 
@mariongalley 

I've reloaded dev (browser cache clear is needed (shift refresh)) and in doing so found a few more cases that needed addressing.

default = award date latest first

**home page**
* When no text in the search field do not change the ordering from the default, check the text input before submitting the search, if empty avoid setting up a "chosen" order-by within the search by redirecting to an empty search results page instead of submitting the form.

**search results page search**
* On an empty search revert to the default ordering
* If a text search is run and then the user delete the contents of the text search unset the best match ordering back to default
* If the search order has been changed by the user after a text search maintain that ordering 
* If an empty search is filtered on with the default ordering and then a text search is added don't override the ordering 
* If the user inputs a text search and then filters do not change ordering from best match
* If the user does a text search and applies a filter, then removes the text search don't reset the search

Commit:
tests: Update for new sort order

- API update aggregates_expected json to change default sort order from
  score to awardDate. These URLs are part of the filter mechanism for
  applying/removing a filter, the sort isn't actually used for API
  purposes.
- Browser tests add new test to check which kind of sort will be applied
  and fix existing test

michaelwood commented:
Note for self re-check if there is something we can do about:

1. Apply Filter "London": sort order is Date
2. Add text "garden": sort switches to Best match
3. Add filter "To individuals": sort stays best match
4. Remove text "garden": sort stay best match - here I think it ought to switch back to date

Expected - revert back to awardDate
